lib/Optimizer/Transforms/DecompositionPatterns.cpp

```diff
@@ -137,6 +137,35 @@ struct HToPhasedRx : public OpRewritePattern<quake::HOp> {
   }
 };
 
+// quake.swap a, b
+// ───────────────────────────────────
+// quake.cnot b, a;
+// quake.cnot a, b;
+// quake.cnot b, a;
+struct SwapToCX : public OpRewritePattern<quake::SwapOp> {
```

Comment:
It would be good to add test with a swap - what was the thing that was failing, and can we please add it as an e2e test?